frontend/src/components/common/Resource/Resource.tsx

```diff
@@ -846,10 +846,48 @@ export function ContainersSection(props: { resource: KubeObjectInterface | null
     return statuses;
   }
 
+  function getEphemeralContainerStatuses() {
+    if (!resource || resource.kind !== 'Pod') {
+      return {};
+    }
+
+    const ephemStatuses: {
+      [key: string]: ContainerInfoProps['status'];
+    } = {};
+
+    ((resource as KubePod).status.ephemeralContainerStatuses || []).forEach(
+      ephemeralContainerStatus => {
+        const { name, ...status } = ephemeralContainerStatus;
+        ephemStatuses[name] = { ...status };
+      }
+    );
+
+    return ephemStatuses;
+  }
+
+  function getInitContainerStatuses() {
```

Comment:
This commit should fix only getInitContainerStatuses and we can reorder it to make it come before the ephemeral container one, so you already have the helper function I mention in the other comment, before you create the getEphemeralContainerStatuses function.